lte/gateway/c/session_manager/LocalEnforcer.cpp

```diff
@@ -1016,13 +1016,44 @@ void LocalEnforcer::init_session(
     const std::string& session_id, const SessionConfig& cfg,
     const CreateSessionResponse& response) {
   const auto time_since_epoch = magma::get_time_in_sec_since_epoch();
-  unsigned shard_id           = shard_tracker_->add_ue(imsi);
   auto session                = std::make_unique<SessionState>(
       imsi, session_id, cfg, *rule_store_, response.tgpp_ctx(),
-      time_since_epoch, response, shard_id);
+      time_since_epoch, response);
   session_map[imsi].push_back(std::move(session));
 }
 
+void LocalEnforcer::add_ue_to_shard(
+    const std::string imsi, SessionMap& session_map, SessionState& session) {
+  // Check if UE has other sessions and reuse shard_id if so
+  auto sm_it     = session_map.find(imsi);
```

Comment:
I think we might have to check that this is a valid iterator right?

Reply:
I can't remember what find returns if the item is not found but let's handle cases in this order:
1. IMSI not in session map -> Is this illegal?
2. Found a different session in session map for the IMSI -> assign shard to new session
3. No other sessions found for the IMSI -> allocate new shard

Reply:
So I can add a check, but there should be no need to check because adding the UE to shard should happen after the session has already been added to the map. To answer those questions:
1. It's illegal.
2. For loop addresses it
3. The last if condition performs the allocation.
I can add further documentation to make all this more clear.

Reply:
My logic for this part is actually incorrect. 
1. It's illegal so shouldn't even be checked.
2. Addressed in for loop
3. Should be addressed at end, but I incorrectly checked if IMSI exists in the map(it always does).